### PR TITLE
Add option to ignore videos when deleting

### PR DESCRIPTION
**Pull request by jonvmey**
Adds a "Delete and Ignore" option when deleting a video which automatically marks it as ignored in the download queue.

Happy to modify the changes if there's a better way to handle this.

Changes related to #286

### CODE REVIEW

File: tubearchivist/static/script.js [hunk 1]
```diff
@@ -381,6 +381,19 @@ function deleteVideo(button) {
   }, 1000);
 }
 
+function deleteAndIgnoreVideo(button) {
+  let to_delete = button.getAttribute('data-id');
+  let to_redirect = button.getAttribute('data-redirect');
+  let endpointDelete = '/api/video/' + to_delete + '/';
+  apiRequest(endpointDelete, 'DELETE');
+  let endpoingDownload = '/api/download/';
```
Comment (barrelltitor) [inline tubearchivist/static/script.js:389]: Here and on the line below shouldn't it be endpoin**t**Download instead of endpoing?

Comment (jonvmey) [reply tubearchivist/static/script.js]: It definitely should.

**Comment by bbilly1** [comment]
Thanks for taking a stab at this. You can even further reuse existing functionality by slightly tweaking the `extrac_dl` task to also take a `status` kwarg, like so:

```python
@shared_task(name="extract_download", bind=True, base=BaseTask)
def extrac_dl(self, youtube_ids, auto_start=False, status="pending"):
    """parse list passed and add to pending"""
    TaskManager().init(self)
    if isinstance(youtube_ids, str):
        to_add = Parser(youtube_ids).parse()
    else:
        to_add = youtube_ids

    pending_handler = PendingList(youtube_ids=to_add, task=self)
    pending_handler.parse_url_list()
    pending_handler.add_to_pending(status=status, auto_start=auto_start)

    if auto_start:
        download_pending.delay(auto_only=True)
```
Then `status` is already implemented in `add_to_pending`. Additionally by handing that off to the task, that can be done async, as that will take a few seconds because it needs to go out to YT to get the data first.

Then from the API side, this would make more sense to implement through the Download Item view `/api/download/<video_id>/` and not the list view, as you are interacting with a single video item, and not with a list. I would suggest to implement this through another `valid_status`, maybe `ignore-force`, to force ignore it even if it doesn't exist? So you can easily separate it with the regular ignore functionality that doesn't need to be called through a task because it's already in the queue.

So partial view something like this:
```python
class DownloadApiView(ApiBaseView):

    valid_status = ["pending", "ignore", "ignore-force", "priority"]

    def post(self, request, video_id):
        """post to video to change status"""
        item_status = request.data.get("status")
        if item_status not in self.valid_status:
            message = f"{video_id}: invalid status {item_status}"
            print(message)
            return Response({"message": message}, status=400)

        if item_status == "ignore-force":
            extrac_dl.delay(video_id, status="ignore")
            message = f"{video_id}: set status to ignore"
            return Response(request.data)

        # rest of the view...

``` 
Then on button click, this would mean two API calls, first to DELETE the video, then POST to `ignore-force` it on the queue, something like this:

```js
function deleteVideo(button) {
  let to_delete = button.getAttribute('data-id');
  let to_ignore = button.getAttribute('data-ignore');
  let to_redirect = button.getAttribute('data-redirect');
  let apiDeleteEndpoint = '/api/video/' + to_delete + '/';
  apiRequest(apiDeleteEndpoint, 'DELETE');

  if (to_ignore !== null) {
    let apiIgnoreEndpoint = '/api/download/' + to_delete + '/';
    apiRequest(apiIgnoreEndpoint, 'POST', { status: 'ignore-force' });
  }

  setTimeout(function () {
    let redirect = '/channel/' + to_redirect;
    window.location.replace(redirect);
  }, 1000);
}
```
While the button would look like this:
```html
<button class="danger-button" onclick="deleteVideo(this)" data-id="{{ video.youtube_id }}" data-ignore data-redirect = "{{ video.channel.channel_id }}">Delete and ignore</button>
```

Does that make sense? Or did I miss something?

Are you up to make these modifications? Otherwise no problem, let me know.

**Comment by jonvmey** [comment]
Thanks for the feedback. I'll look over your suggestions and work on implementing them.

**Comment by bbilly1** [comment]
are you still looking into this? Otherwise let me know.

**Comment by bbilly1** [comment]
closing this in favor of: https://github.com/tubearchivist/tubearchivist/commit/852abf254d0dac5c5b4265df3285218559a943ba

**Pull request closed without merging**